actors/datacap/src/lib.rs

```diff
@@ -465,86 +466,30 @@ impl<T> AsActorResult<T> for Result<T, TokenError> {
-                serialize(&ret, "mint result")
-            }
-            Some(Method::DestroyExported) => {
-                let ret = Self::destroy(rt, cbor::deserialize_params(params)?)?;
-                serialize(&ret, "destroy result")
-            }
-            Some(Method::NameExported) => {
-                let ret = Self::name(rt)?;
-                serialize(&ret, "name result")
-            }
-            Some(Method::SymbolExported) => {
-                let ret = Self::symbol(rt)?;
-                serialize(&ret, "symbol result")
-            }
-            Some(Method::GranularityExported) => {
-                let ret = Self::granularity(rt)?;
-                serialize(&ret, "granularity result")
-            }
-            Some(Method::TotalSupplyExported) => {
-                let ret = Self::total_supply(rt, cbor::deserialize_params(params)?)?;
-                serialize(&ret, "total_supply result")
-            }
-            Some(Method::BalanceExported) => {
-                let ret = Self::balance(rt, cbor::deserialize_params(params)?)?;
-                serialize(&ret, "balance_of result")
-            }
-            Some(Method::TransferExported) => {
-                let ret = Self::transfer(rt, cbor::deserialize_params(params)?)?;
-                serialize(&ret, "transfer result")
-            }
-            Some(Method::TransferFromExported) => {
-                let ret = Self::transfer_from(rt, cbor::deserialize_params(params)?)?;
-                serialize(&ret, "transfer_from result")
-            }
-            Some(Method::IncreaseAllowanceExported) => {
-                let ret = Self::increase_allowance(rt, cbor::deserialize_params(params)?)?;
-                serialize(&ret, "increase_allowance result")
-            }
-            Some(Method::DecreaseAllowanceExported) => {
-                let ret = Self::decrease_allowance(rt, cbor::deserialize_params(params)?)?;
-                serialize(&ret, "decrease_allowance result")
-            }
-            Some(Method::RevokeAllowanceExported) => {
-                Self::revoke_allowance(rt, cbor::deserialize_params(params)?)?;
-                Ok(RawBytes::default())
-            }
-            Some(Method::BurnExported) => {
-                let ret = Self::burn(rt, cbor::deserialize_params(params)?)?;
-                serialize(&ret, "burn result")
-            }
-            Some(Method::BurnFromExported) => {
-                let ret = Self::burn_from(rt, cbor::deserialize_params(params)?)?;
-                serialize(&ret, "burn_from result")
-            }
-            Some(Method::AllowanceExported) => {
-                let ret = Self::allowance(rt, cbor::deserialize_params(params)?)?;
-                serialize(&ret, "allowance result")
-            }
-            None => Err(actor_error!(unhandled_message; "Invalid method")),
-        }
+    type Methods = Method;
+    actor_dispatch_restricted! {
```

Comment:
I would consider un-renaming this to reduce conflicts. I.e., `actor_dispatch` and `actor_dispatch_unrestricted`.

Reply:
Sure, happy to do so.